src/Repository/EventRepository.php

```diff
@@ -113,4 +113,149 @@ public function countItemsAdmin()
+        $qb = $this->setCriterias($params);
+
+        return $qb
+            ->select('count(event.id)')
+            ->getQuery()
+            ->getSingleScalarResult();
+    }
+
+    public function setCriterias($params)
+    {
+        $qb = $this->createQueryBuilder('event')
+            ->innerJoin('event.subgenre', 'subgenre')
+            ->innerJoin('subgenre.genre', 'genre')
+            ->leftJoin('event.weapons', 'weapon');
+
+        if (count($params['type'])>0) {
+            $entityList = [];
+            foreach ($params['type'] as $entity) {
+                $entityList[] = $entity->getId();
+            }
+            $qb->andWhere($qb->expr()->in('event.type', $entityList));
+        }
+
+        if (count($params['region'])>0) {
+            $entityList = [];
+            foreach ($params['region'] as $entity) {
+                $entityList[] = $entity->getId();
+            }
+            $qb->andWhere($qb->expr()->in('event.region', $entityList));
+        }
+
+        if (count($params['genre'])>0) {
+            $entityList = [];
+            foreach ($params['genre'] as $entity) {
+                $entityList[] = $entity->getId();
+            }
+            dump($entityList);
+            $qb->andWhere($qb->expr()->in('genre', $entityList));
+        }
+
+        if (count($params['subgenre'])>0) {
+            $entityList = [];
+            foreach ($params['subgenre'] as $entity) {
+                $entityList[] = $entity->getId();
+            }
+            $qb->andWhere($qb->expr()->in('event.subgenre', $entityList));
+        }
+
+        if (count($params['settlement'])>0) {
+            $entityList = [];
+            foreach ($params['settlement'] as $entity) {
+                $entityList[] = $entity->getId();
+            }
+            $qb->andWhere($qb->expr()->in('event.settlement', $entityList));
+        }
+
+        if (count($params['weapons'])>0) {
+            $entityList = [];
+            foreach ($params['weapons'] as $entity) {
+                $entityList[] = $entity->getId();
+            }
+            $qb->andWhere($qb->expr()->in('weapon', $entityList));
+        }
+
+        if ($params['priceMax']) {
+            $qb->andWhere($qb->expr()->lte('event.priceMin', $params['priceMax']));
+        }
+
+        if ($params['keywords']) {
+            $words = explode(" ", $params['keywords']);
+
+            $conditions = [];
+            foreach ($words as $word) {
+                $conditions[] = $qb->expr()->like('event.name', $qb->expr()->literal('%' . $word . '%'));
+                $conditions[] = $qb->expr()->like('event.description', $qb->expr()->literal('%' . $word . '%'));
+            }
+            $qb->andWhere(
+                call_user_func_array(array($qb->expr(), 'orX'), $conditions)
+            );
+        }
+
+        $qb
+            ->andWhere('event.startDate >= :start')
+            ->andWhere('event.startDate <= :end')
+            ->andWhere('event.status = :status')
```

Comment:
@daria-sieroshtan 
Use expressions, Luke
`$qb->expr()->gte()` (**g**reater **t**han or **e**quals)
`$qb->expr()->lte()`
`$qb->expr()->eq()`

Reply:
There will be additional dances with tambourine for date conversion, expr()->literal() and so on. I prefer to keep it as is.

Reply:
@daria-sieroshtan 
Expression "dances with tambourine" is reserved for doing weird stuff.
Using expressions here would not be weird - in fact, this is exactly how it should be done. If you have troubles grasping this, those troubles are another reason to practise more

Reply:
PS. You don't need literals here